ocaml/xapi/xapi_http.ml

```diff
@@ -199,7 +213,11 @@ let with_context ?(dummy = false) label (req : Request.t) (s : Unix.file_descr)
         | None, None, Some (Http.UnknownAuth x) ->
             raise (Failure (Printf.sprintf "Unknown authorization header: %s" x))
         | None, None, None ->
-            raise (Http.Unauthorised label)
+            let session_id =
```

Comment:
While this addition is probably fine by itself, I feel this entire function is increasingly getting out of shape. We have deeply nested pattern matching and exception handling and I believe this is in need of a rethink.

Reply:
I appended a refine commit for this. Hope this would be better.